TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/DateDueScreen/DateDueScreenViewController.m

```diff
@@ -0,0 +1,58 @@
+//
+//  DateDueScreenViewController.m
+//  TestAssignmentDoItSoftware
+//
+//  Created by Vladimir Samoylenko on 2/18/19.
+//  Copyright © 2019 TheRedQueen. All rights reserved.
+//
+
+#import "DateDueScreenViewController.h"
+
+@interface DateDueScreenViewController ()
+
+@end
+
+@implementation DateDueScreenViewController
+
+
+@synthesize btnSetDateDueForTask;
+@synthesize dpDateDue;
+
+
+- (void)viewDidLoad {
+    
+    [super viewDidLoad];
+
+    self.navigationController.navigationBar.topItem.title = @"Back";
+    
+    self.navigationItem.title = @"Edit task";
+    self.navigationController.navigationBar.translucent = NO;
+    self.navigationController.navigationBar.barTintColor = [UIColor colorWithRed:232.0/255.0 green:232.0/255.0 blue:232.0/255.0 alpha:1.0];
+    self.navigationController.navigationBar.tintColor = [UIColor colorWithRed:130.0/255.0 green:130.0/255.0 blue:130.0/255.0 alpha:1.0];
+
+}
+
+
+- (IBAction)btnSetDateDueForTask:(id)sender{
+    
+#warning add check if date is in the past login has not been provided in description so it not implemented
+    
+    NSLog(@"set due date tapped");
+
+    NSDate *dateMoment = dpDateDue.date;
+    NSNumber *dueDateMoment = [NSNumber numberWithUnsignedInteger:[dateMoment timeIntervalSince1970]];
+    
+    NSLog(@"date moment is: %ld", (long)[dueDateMoment unsignedIntegerValue]);
+
+    
+    NSArray *viewControllers = self.navigationController.viewControllers;
+    EditTaskScreenViewController *editTaskScreen = [viewControllers objectAtIndex:self.navigationController.viewControllers.count - 2];
```

Comment:
Never set expectations on the navigation stack. Use notifications, delegates, callbacks, or any other means of indirect data transfer.

Reply:
One option would be to use a delegate:

```
if (dueDateDelegate) {
  [dueDateDelegate dueDateChanged:dueDateMoment];
}
```

where `dueDateDelegate` would be a weak property conforming to

```
@protocol DueDateDelegate
- (void)dueDateChanged:(NSNumber)date;
@end
```

It'd be set by the caller before pushing this controller on the stack.